tsl/src/compression/compression.c

```diff
@@ -2272,6 +2272,97 @@ ts_read_compressed_data_directory(PG_FUNCTION_ARGS)
 
 #endif
 
+#ifdef TS_COMPRESSION_FUZZING
+
+/*
+ * This is our test function that will be called by the libfuzzer driver. It
+ * has to catch the postgres exceptions normally produced for corrupt data.
+ */
+static int
+llvm_fuzz_target_generic(int (*target)(const uint8_t *Data, size_t Size, bool check_compression),
+						 const uint8_t *Data, size_t Size)
+{
+	MemoryContextReset(CurrentMemoryContext);
+
+	PG_TRY();
+	{
+		CHECK_FOR_INTERRUPTS();
+		target(Data, Size, /* check_compression = */ false);
+	}
+	PG_CATCH();
+	{
+		FlushErrorState();
+	}
+	PG_END_TRY();
+
+	/* We always return 0, and -1 would mean "don't include it into corpus". */
+	return 0;
+}
+
+static int
+llvm_fuzz_target_gorilla_float8(const uint8_t *Data, size_t Size)
+{
+	return llvm_fuzz_target_generic(decompress_gorilla_float8, Data, Size);
+}
+static int
+llvm_fuzz_target_deltadelta_int64(const uint8_t *Data, size_t Size)
+{
+	return llvm_fuzz_target_generic(decompress_deltadelta_int64, Data, Size);
+}
+
+/*
+ * This is the libfuzzer fuzzing driver. It will run our test functions with
+ * random inputs.
+ */
+extern int LLVMFuzzerRunDriver(int *argc, char ***argv,
+							   int (*UserCb)(const uint8_t *Data, size_t Size));
+
+TS_FUNCTION_INFO_V1(ts_fuzz_compression);
+
+Datum
+ts_fuzz_compression(PG_FUNCTION_ARGS)
+{
+	MemoryContext fuzzing_context =
```

Comment:
Could you add a comment here on why this specific size of the memory context was chosen?

Reply:
Added. Basically it's just an arbitrary large value to limit the memory context to one chunk and avoid reallocation of chunks after it is reset.